packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3646,10 +3646,14 @@ export class ContainerRuntime
 
 	public notifyAttaching() {} // do nothing (deprecated method)
 
-	public getPendingLocalState(): unknown {
+	public async getPendingLocalState(close?: boolean): Promise<unknown> {
+		this.verifyNotClosed();
 		if (this._orderSequentiallyCalls !== 0) {
 			throw new UsageError("can't get state during orderSequentially");
 		}
+		if (close) {
```

Comment:
does this need to be a new call, or can it just go inside this.blobManager.getPendingBlobs() which would then become async. i think that would be a better encapsulation of the feature, rather than forcing the user to do it.

Reply:
I'm not sure if we want to shutdown blob processing if we're not closing the container.

Reply:
we don't, the getPendingBlobs will also take the props object

Reply:
what about getLocalState? we need to wait for shutdown before calling it.